ui-particles-angular/projects/ui-particles-angular/src/lib/gio-menu/gio-menu-item/gio-menu-item.component.ts

```diff
@@ -24,4 +26,26 @@ export class GioMenuItemComponent {
   @Input() public icon = '';
   @Input() public active = false;
   @Input() public outlined = false;
+
+  constructor(private readonly gioMenuService: GioMenuService) {}
+
+  public onMouseLeave($event: MouseEvent): void {
+    if (this.active) {
+      const target = $event.target as HTMLInputElement;
+      const menuItem = target.closest('.gio-menu-item') as HTMLInputElement;
+      setTimeout(() => this.gioMenuService.mouseOverItem({ enter: false, top: menuItem.getBoundingClientRect().top }), 50);
+    }
+  }
+
+  public onMouseEnter($event: MouseEvent): void {
+    if (this.active) {
+      const target = $event.target as HTMLInputElement;
+      const menuItem = target.closest('.gio-menu-item') as HTMLInputElement;
+      this.gioMenuService.mouseOverItem({ enter: true, top: menuItem.getBoundingClientRect().top });
+    }
+  }
+
+  public onClick($event: MouseEvent): void {
+    setTimeout(() => this.onMouseEnter($event), 0);
```

Comment:
the setTimeout with 0n value is required ?

Reply:
Yes so that it runs at the end of the stack